Reduce rerenders in the metadata tab

Arnei opened the pull request:
This patch rewrites the Metadata component. This accomplishes:
- Stop the whole Metadata component from rerendering each time redux state is updated. This was causing ui elements to become unresponsive in certain cases (e.g. having tens of thousands of series in your Opencast).
- Makes the code in the Metadata component far less difficult to read..

### How to test this

Can be tested as usual. Make sure that editing and saving Metadata still works as expected.

Arnei committed:
Reduce rerenders in the metadata tab

This patch rewrites the Metadata component. This accomplishes:
- Stop the whole Metadata component from rerendering
each time redux state is updated. This was causing ui elements to
become unresponsive in certain cases (e.g. having tens of thousands of
series in your Opencast).
- Makes the code in the Metadata component far less
difficult to read.

Arnei committed:
Merge remote-tracking branch 'upstream/r/18.x' into metadata-rework-reduce-rerenders

gregorydlogan commented:
Seeing an exception when changing the start date:

```
2025-10-20 11:28:11,515 | INFO  | (AssetManagerImpl:452) - Trigger update handlers for snapshot ID-dog-rose-1310279468, version 3
2025-10-20 11:28:11,574 | ERROR | (MetadataJson:365) - Not able to parse date string 2021-06-15T13:33: ParseException: Unparseable date: "2021-06-15T13:33"
2025-10-20 11:28:11,575 | ERROR | (MetadataJson:283) - Not able to parse date 2021-06-14T13:33: Unparseable date: "2021-06-14T13:33"
2025-10-20 11:28:11,608 | INFO  | (AssetManagerImpl:1321) - Creating new version 4 of media package ID-dog-rose-1310279468

```

No present without this PR.

Arnei committed:
Fix metadata date not being updated

The backend would report
"Not able to parse date string 2021-06-15T13:33: ParseException: Unparseable date: "2021-06-15T13:33"".

This fixes that, by making sure the date stored in redux is
a proper ISO date again.
